python/pyspark/sql/column.py

```diff
@@ -175,46 +175,13 @@ def _bin_op(
     ["Column", Union["Column", "LiteralType", "DecimalLiteral", "DateTimeLiteral"]], "Column"
 ]:
     """Create a method for given binary operator"""
-    binary_operator_map = {
-        "plus": "+",
-        "minus": "-",
-        "divide": "/",
-        "multiply": "*",
-        "mod": "%",
-        "equalTo": "=",
-        "lt": "<",
-        "leq": "<=",
-        "geq": ">=",
-        "gt": ">",
-        "eqNullSafe": "<=>",
-        "bitwiseOR": "|",
-        "bitwiseAND": "&",
-        "bitwiseXOR": "^",
-        # Just following JVM rule even if the names of source and target are the same.
-        "and": "and",
-        "or": "or",
-    }
 
     def _(
         self: "Column",
         other: Union["Column", "LiteralType", "DecimalLiteral", "DateTimeLiteral"],
     ) -> "Column":
         jc = other._jc if isinstance(other, Column) else other
-        if name in binary_operator_map:
-            from pyspark.sql import SparkSession
-
-            spark = SparkSession._getActiveSessionOrCreate()
-            stack = list(reversed(inspect.stack()))
-            depth = int(
-                spark.conf.get("spark.sql.stackTracesInDataFrameContext")  # type: ignore[arg-type]
```

Comment:
shall we still respect this config in the new code?

Reply:
Oh, yes we should. Just updated code to respect the config.